src/rp2/plugin/report/us/open_positions.py

```diff
@@ -0,0 +1,233 @@
+
+        # pylint: disable=too-many-branches
+
+        sheet_names: Set[str] = {"Asset", "Asset_and_Exchange", "AssetPrice"}
+        row_indexes: Dict[str, int] = {sheet_name: self.HEADER_ROWS for sheet_name in sheet_names}
+
+        if not isinstance(asset_to_computed_data, Dict):
+            raise RP2TypeError(f"Parameter 'asset_to_computed_data' has non-Dict value {asset_to_computed_data}")
+
+        output_file: Any
+        output_file = self._initialize_output_file(
+            country=country,
+            accounting_method=accounting_method,
+            output_dir_path=output_dir_path,
+            output_file_prefix=output_file_prefix,
+            output_file_name=self.OUTPUT_FILE,
+            template_sheets_to_keep=_TEMPLATE_SHEETS_TO_KEEP,
+            from_date=from_date,
+            to_date=to_date,
+            template_file_prefix="open_positions",
+        )
+
+        asset: str
+        computed_data: ComputedData
+
+        ap_sheet = output_file.sheets["AssetPrice"]
+        ae_sheet = output_file.sheets["Asset_and_Exchange"]
+        a_sheet = output_file.sheets["Asset"]
+
+        # First loop through, I want to know the total cost basis for all assets and exchanges after filtering out net cost under $0.01.
+        total_cost: RP2Decimal = RP2Decimal("0")
+        for asset, computed_data in asset_to_computed_data.items():
+            if not isinstance(asset, str):
+                raise RP2TypeError(f"Parameter 'asset' has non-string value {asset}")
+            ComputedData.type_check("computed_data", computed_data)
+
+            net_cost = RP2Decimal("0")
+            for current_transaction in computed_data.in_transaction_set:
+                in_xact = cast(InTransaction, current_transaction)
+                net_cost += in_xact.fiat_in_with_fee * (RP2Decimal("1") - computed_data.get_in_lot_sold_percentage(in_xact))
+
+            if net_cost < _FIAT_EMPTY_BAL_CUTOFF:
+                continue
+
+            total_cost += net_cost
+
+        # Loop thru every asset again, but this time we can do reporting.
+        for asset, computed_data in asset_to_computed_data.items():
+            if not isinstance(asset, str):
+                raise RP2TypeError(f"Parameter 'asset' has non-string value {asset}")
+            ComputedData.type_check("computed_data", computed_data)
+
+            # Net cost basis will be the sum of all usd_in_w_fee for an asset after multiplying off the sold percentage.
+            net_cost = RP2Decimal("0")
+            for current_transaction in computed_data.in_transaction_set:
+                in_xact = cast(InTransaction, current_transaction)
+                sold_pct: RP2Decimal = computed_data.get_in_lot_sold_percentage(in_xact)
+                net_cost += in_xact.fiat_in_with_fee * (RP2Decimal("1") - sold_pct)
+
+            # Sum the total amount of crypto asset remaining.
+            tot_crypto_bal = RP2Decimal("0")
+            for balset in computed_data.balance_set:
```

Comment:
Fixed the varname. Will have to spend a little extra time making this work per user. Can we make that a separate PR or would you prefer leaving it as part of this one?

Reply:
I think it's worth spending a little extra time on this one, otherwise people filing together get confusing results.

Reply:
That does make sense. I've started refactoring the plugin to account for this. 

I set up a test with two users, e.g. a second user buys, receives a transfer, and sells BTC interspersed with the first user's activity. It appears as though the final balance on the tax form accurately tracks the balance for each user, but sales (at least in a FIFO scenario) are top-down from the combined assets rather than e.g. in processing User2's sale first hunting to their buy and drawing from that purchase first. 

I just want to confirm this is expected - if so, it does make adding multiple users to my report easier - as I can keep total cost basis as the sum for all holders of an asset and then divide that out over their respective final balances. 

My original concern was that this wouldn't be purely representational if they were individuals tracking their cost basis, but if the above is accurate, then it's OK that the holders all have the same unit cost basis, just with different crypto balances / allocations.

Can you confirm the function? Hopefully what I'm asking makes sense. :)

Reply:
Yes, that sounds correct: they are filing together so their per-unit cost basis is the same but they have different balances.

Reply:
Changes pushed incorporating multiple holders and other changes previously marked resolved since last review.